asio: 1.24.0 -> 1.36.0

Aleksanaa opened the pull request:
http://asio.sourceforge.net/ redirects to https://think-async.com/Asio/
and https://think-async.com/Asio/Repository.html points to new GitHub
repository. So this should be the new upstream, not another fork.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Aleksanaa committed:
asio_1_10: drop

Aleksanaa committed:
asio: move to by-name

Aleksanaa committed:
asio: 1.24.0 -> 1.36.0

http://asio.sourceforge.net/ redirects to https://think-async.com/Asio/
and https://think-async.com/Asio/Repository.html points to new GitHub
repository. So this should be the new upstream, not another fork.

Aleksanaa committed:
asio: add aleksana to maintainers

panicgh reviewed (commented):
asio 1.34.2 is apparently too new for many pkgs. In many cases porting would be almost trivial (e.g. renames of certain types and functions), but apparently it was not yet done in those projects.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 414478`
Commit: `ae3dae5b1e341a98d2868440d111a1fc7791ff3d`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>obs-studio-plugins.obs-nvfbc</li>
  </ul>
</details>
<details>
  <summary>:x: 73 packages failed to build:</summary>
  <ul>
    <li>crow</li>
    <li>freedv</li>
    <li>hidviz</li>
    <li>hpx</li>
    <li>libpulsar</li>
    <li>mariadb-galera</li>
    <li>musikcube</li>
    <li>musikcube.dev</li>
    <li>nuraft</li>
    <li>obs-studio</li>
    <li>obs-studio-plugins.advanced-scene-switcher</li>
    <li>obs-studio-plugins.distroav</li>
    <li>obs-studio-plugins.droidcam-obs</li>
    <li>obs-studio-plugins.input-overlay</li>
    <li>obs-studio-plugins.looking-glass-obs</li>
    <li>obs-studio-plugins.obs-3d-effect</li>
    <li>obs-studio-plugins.obs-advanced-masks</li>
    <li>obs-studio-plugins.obs-aitum-multistream</li>
    <li>obs-studio-plugins.obs-backgroundremoval</li>
    <li>obs-studio-plugins.obs-browser-transition</li>
    <li>obs-studio-plugins.obs-color-monitor</li>
    <li>obs-studio-plugins.obs-command-source</li>
    <li>obs-studio-plugins.obs-composite-blur</li>
    <li>obs-studio-plugins.obs-dir-watch-media</li>
    <li>obs-studio-plugins.obs-dvd-screensaver</li>
    <li>obs-studio-plugins.obs-freeze-filter</li>
    <li>obs-studio-plugins.obs-gradient-source</li>
    <li>obs-studio-plugins.obs-gstreamer</li>
    <li>obs-studio-plugins.obs-hyperion</li>
    <li>obs-studio-plugins.obs-livesplit-one</li>
    <li>obs-studio-plugins.obs-markdown</li>
    <li>obs-studio-plugins.obs-media-controls</li>
    <li>obs-studio-plugins.obs-move-transition</li>
    <li>obs-studio-plugins.obs-multi-rtmp</li>
    <li>obs-studio-plugins.obs-mute-filter</li>
    <li>obs-studio-plugins.obs-ndi</li>
    <li>obs-studio-plugins.obs-noise</li>
    <li>obs-studio-plugins.obs-pipewire-audio-capture</li>
    <li>obs-studio-plugins.obs-recursion-effect</li>
    <li>obs-studio-plugins.obs-replay-source</li>
    <li>obs-studio-plugins.obs-retro-effects</li>
    <li>obs-studio-plugins.obs-rgb-levels</li>
    <li>obs-studio-plugins.obs-scale-to-sound</li>
    <li>obs-studio-plugins.obs-scene-as-transition</li>
    <li>obs-studio-plugins.obs-shaderfilter</li>
    <li>obs-studio-plugins.obs-source-clone</li>
    <li>obs-studio-plugins.obs-source-record</li>
    <li>obs-studio-plugins.obs-source-switcher</li>
    <li>obs-studio-plugins.obs-stroke-glow-shadow</li>
    <li>obs-studio-plugins.obs-teleport</li>
    <li>obs-studio-plugins.obs-text-pthread</li>
    <li>obs-studio-plugins.obs-transition-table</li>
    <li>obs-studio-plugins.obs-tuna</li>
    <li>obs-studio-plugins.obs-urlsource</li>
    <li>obs-studio-plugins.obs-vaapi</li>
    <li>obs-studio-plugins.obs-vertical-canvas</li>
    <li>obs-studio-plugins.obs-vintage-filter</li>
    <li>obs-studio-plugins.obs-vkcapture</li>
    <li>obs-studio-plugins.obs-vnc</li>
    <li>obs-studio-plugins.obs-webkitgtk</li>
    <li>obs-studio-plugins.obs-websocket</li>
    <li>obs-studio-plugins.pixel-art</li>
    <li>obs-studio-plugins.waveform</li>
    <li>obs-studio-plugins.wlrobs</li>
    <li>obs-studio.debug</li>
    <li>openvpn3</li>
    <li>pc-ble-driver</li>
    <li>python312Packages.pulsar-client</li>
    <li>python312Packages.pulsar-client.dist</li>
    <li>python313Packages.pulsar-client</li>
    <li>python313Packages.pulsar-client.dist</li>
    <li>sioclient</li>
    <li>widelands</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>asio</li>
    <li>gigedit</li>
    <li>ioq3-scion</li>
    <li>jami</li>
    <li>linuxsampler</li>
    <li>maa-assistant-arknights</li>
    <li>maa-cli</li>
    <li>opendht</li>
    <li>opendht.dev</li>
    <li>opendht.lib</li>
    <li>opendht.man</li>
    <li>pan-bindings</li>
    <li>restinio</li>
    <li>snapcast</li>
  </ul>
</details>

Aleksanaa commented on pkgs/by-name/as/asio/package.nix:
```diff
@@ -1,29 +1,50 @@
+  src = fetchFromGitHub {
+    owner = "chriskohlhoff";
+    repo = "asio";
+    tag = "asio-${lib.replaceStrings [ "." ] [ "-" ] finalAttrs.version}";
```
fyi `lib.replaceStrings` is an alias for `builtins.replaceStrings` for better compatibility

Aleksanaa commented on pkgs/by-name/as/asio/package.nix:
```diff
@@ -0,0 +1,65 @@
+  # Only used for test coverage
+  checkInputs = [
+    openssl
+    boost
```
I decided to stop propagating boost because it seems that asio headers can change whether to use boost or std namespace according to the definition of macros, and downstream consumers may not like the boost version we provide.

I also tried various `--with-boost`, and I'm very sure asio found the boost header (`checking for boost/noncopyable.hpp... yes`), but it didn't affect our output.

At the same time, considering that people who use asio may want to avoid boost, let's not propagate it by default.

Aleksanaa committed:
websocketpp: add patch to fix build with boost187

Aleksanaa commented:
obs-studio builds now!

Aleksanaa committed:
freedv: remove sioclient dependency

This is to prepare for the removal of sioclient, which doesn't build
with newer asio. Freedv no longer depends on sioclient after 2.0.0,
which got removed in https://github.com/drowe67/freedv-gui/pull/836

Aleksanaa committed:
sioclient: drop

Aleksanaa committed:
hpx: add patch to fix build with asio >= 1.34.0

Aleksanaa committed:
libpulsar: pin asio or boost version

To fix build after io_service removal in asio (as well as boost::asio)

Aleksanaa committed:
crow: 1.2.1.2 -> 1.3.0.0

Aleksanaa committed:
hidviz: pin to asio_1_32_0

Aleksanaa committed:
mariadb-galera: pin to asio_1_32_0

Aleksanaa committed:
nuraft: pin to asio_1_32_0

Aleksanaa committed:
openvpn3: pin to asio_1_32_0

Aleksanaa committed:
pc-ble-driver: pin to asio_1_32_0

Aleksanaa committed:
doc/release-notes/rl-2605: add notice about asio update

Aleksanaa commented:
> ioq3-scion
> libpulsar

These two fail on hydra too.

https://hydra.nixos.org/build/314374688
https://hydra.nixos.org/build/314383010



> pc-ble-driver

Yeah, forgot to note, this also fails for me on x86_64-linux even before the change. I just saw related error messages after update, so I fixed my part but it still doesn't build.
